cicd/gitlab/parts/python.gitlab-ci.yml

```diff
@@ -0,0 +1,107 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Python related part of the Ensembl/ensembl-genomio CI/CD pipeline
+
+variables:
+  # legacy, used to find ensembl-hive python parts
+  PYTHONPATH: $RUN_DIR/ensembl-hive/wrappers/python3
+
+# don't fotget to update /pyproject.toml
+#  if changing python version
+image: python:3.8
+
+
+# overriding the default stages  list (https://docs.gitlab.com/ee/ci/yaml/#stages)
+stages:
+  - python:prepare_env
+  - python:static_analysis
+
+
+# Python related environmnet and tests
+python:prepare_env:venv:
+  stage: python:prepare_env
+  artifacts:
+    paths:
+      - $RUN_DIR
+  before_script:
+    - python --version # For debugging
+    - python -m venv $RUN_DIR/venv
+    - source $RUN_DIR/venv/bin/activate
+  script:
+    - pip install mypy
+    - pip install types-requests  # type stub for "requests" (or run `mypy --install-types`)
+    - pip install pylint
+    - pip install black
+
+
+##  LOCAL ENV & PIP
+python:prepare_env:local:
+  stage: python:prepare_env
+  needs:
+    # using `needs` as it allows to depend on jobs from the same stage
+    #  https://docs.gitlab.com/ee/ci/yaml/#needs
+    - job: python:prepare_env:venv
+      artifacts: true
+  artifacts:
+    paths:
+      - $RUN_DIR
+  before_script:
+    - source $RUN_DIR/venv/bin/activate
+  script:
+    # pip based
+    - pip install -e .[dev]
```

Comment:
Since this covers the dependencies installed before (`python:prepare_env:venv:`), is there a need to have these 2 separated? I would expect all testing tasks (`pylint`, `mypy`, `pytest`, ...) to depend on this last job rather than the previous one.

Reply:
I'd rather prefer to have this as a separate task.  The previous one is more about tools and external bits installation